Intel vRAN Boost operator (#249)
StarlingXWeb/starlingx-website: src/pages/blog/starlingx-intel-vran-boost.md

---
templateKey: blog-post
title: 4th Gen Intel Xeon Scalable Processor with Intel vRAN Boost support in StarlingX
author: Balendu Mouli Burla
category: 
  - label: Features & Updates
---
Learn more about Intel vRAN Boost Kubernetes operator support in StarlingX 9.0. <!-- more -->

The [StarlingX 9.0](https://www.starlingx.io/blog/starlingx-release-9/) release includes a new Kubernetes Operator to manage and configure the integrated Intel vRAN Boost accelerator in 4th Gen Intel Xeon Scalable processors. In this blog post, you will learn about Intel vRAN Boost Kubernetes Operator functionality in StarlingX.

> As communications service providers move from 4G to 5G networks, many Telecommunication providers developing virtualized Radio Access Network (vRAN) architecture for higher channel capacity and easier deployment of Edge based services and applications. vRAN solutions are ideally located to deliver low latency services with flexibility to increase or decrease capacity based on the volume of real time traffic and demand on the network with a great support from Operating System and underlying hardware platform.

Reviewer: ```suggestion
As communications service providers move from 4G to 5G networks, many telecom operators are developing virtualized Radio Access Network (vRAN) architecture for higher channel capacity and easier deployment of Edge based services and applications. vRAN solutions are ideally located to deliver low latency services with flexibility to increase or decrease capacity based on the volume of real time traffic and demand on the network with a great support from the operating system and underlying hardware platform.
```

Intel vRAN Boost is an integrated vRAN Accelerator in [4th Gen Intel Xeon Scalable processors](https://www.intel.com/content/www/us/en/products/docs/processors/xeon-accelerated/4th-gen-xeon-scalable-processors-product-brief.html) to provide outstanding energy-efficient performance across networking use cases, including vRAN deployments.

### Intel vRAN Boost Kubernetes Operator 

> Intel vRAN Boost Kubernetes Operator provides the ability to orchestrate, configure and manage the Intel vRAN Boost integrated accelerator in latest 4th Generation Intel Xeon Scalable processors when deployed on a Kubernetes cluster(s). The Operator is backward compatible, it includes support for previous generations of Intel vRAN Accelerator ACC100 Adapter cards as well. The Operator is a state machine which configures the resources, monitors them, and acts autonomously based on the user interaction. Operators perform the functionality of

Reviewer: ```suggestion
Intel vRAN Boost Kubernetes Operator provides the ability to orchestrate, configure and manage the Intel vRAN Boost integrated accelerator in latest 4th Generation Intel Xeon Scalable processors when deployed on a Kubernetes cluster(s). The Operator is backward compatible, it includes support for previous generations of Intel vRAN Accelerator ACC100 Adapter cards as well. The Operator is a state machine which configures the resources, monitors them, and acts autonomously based on the user interaction. The Operator can perform the functionality of
```

> - Detecting and labeling the worker nodes which has accelerators within the Kubernetes cluster.

Reviewer: ```suggestion
- Detecting and labeling the worker nodes which have accelerators within the Kubernetes cluster.
```

- Configuring the accelerators as per vRAN deployment requirements.

> - Creating the Virtual Functions (VFs) from accelerator. 

Reviewer: ```suggestion
- Creating the Virtual Functions (VFs) for the accelerator(s). 
```

> - Maintaining VFs as Kubernetes resources and manage the resource allocations to vRAN applications.

Reviewer: ```suggestion
- Maintaining VFs as Kubernetes resources and managing the resource allocations to vRAN applications.
```

- Providing user/admin interface access to the accelerator through Kubernetes Custom Resource (CRs) APIs.
- Managing the Accelerator resource in Secure boot environments as well.

 ### Intel vRAN Boost Kubernetes Operator integrated with StarlingX 9.0

> The Intel vRAN Boost Kubernetes Operator is integrated in StarlingX as a system application. With the [StarlingX Application Package Manager](https://docs.starlingx.io/system_configuration/kubernetes/system-configuration-starlingx-application-package-manager.html), the Intel vRAN Boost Kubernetes Operator application’s lifecycle can be managed, including managing over-rides to the helm charts for controlling the operator behavior. The latest version of the Operator supported in StarlingX 9.0 is v2.9.0.

Reviewer: ```suggestion
The Intel vRAN Boost Kubernetes Operator is integrated in StarlingX as a system application. With the [StarlingX Application Package Manager](https://docs.starlingx.io/system_configuration/kubernetes/system-configuration-starlingx-application-package-manager.html), the Intel vRAN Boost Kubernetes Operator application’s lifecycle can be managed, including managing overrides to the helm charts for controlling the operator behavior. The latest version of the Operator supported in StarlingX 9.0 is v2.9.0.
```

> A StarlingX application package is a compressed tarball containing a metadata.yaml file, a manifest.yaml, FluxCD manifest files, and a charts directory containing helm charts and a checksum.md5 file. The metadata.yaml file contains references to the application name, version, as well as references to StarlingX maintained docker images for Intel vRAN Boost Kubernetes Operator.

Reviewer: ```suggestion
A StarlingX application package is a compressed tarball containing the following:
- metadata.yaml
- manifest.yaml
- FluxCD manifest files
- checksum.md5
- charts directory containing helm charts 

The metadata.yaml file contains references to the application name, version, as well as references to StarlingX maintained docker images for Intel vRAN Boost Kubernetes Operator.
```

---
templateKey: blog-post

> title: 4<sup>th</sup> Gen Intel Xeon Scalable Processor with Intel vRAN Boost support in StarlingX

Author: Remove, HTML all from all the places. thanks.

---
templateKey: blog-post
title: 4<sup>th</sup> Gen Intel Xeon Scalable Processor with Intel vRAN Boost support in StarlingX
author: Balendu Mouli Burla
category: 
  - label: Features & Updates
---
Learn more about Intel vRAN Boost Kubernetes operator support in StarlingX 8.0. <!-- more -->

The [StarlingX 8.0](https://www.starlingx.io/blog/starlingx-release-8/) release includes a new Kubernetes Operator to manage and configure the integrated Intel vRAN Boost accelerator in 4<sup>th</sup> Gen Intel Xeon Scalable processors. In this blog post, we will learn about Intel vRAN Boost Kubernetes Operator functionality in StarlingX.

> 

Reviewer: The first two sections somewhat read as a sales pitch, due to their focus on the specific capabilities of the 4<sup>th</sup> Gen Intel Xeon Scalable processors.  To make the content a better fit for the StarlingX community blog, it would be great to describe the processor's capabilities in the context of the StarlingX platform.

I would like to suggest the following approach:

- Start with a brief introduction about vRAN. Please also resolve the acronym for those, who might be new to that use case.
- Continue with the 'Intel vRAN Boost Kubernetes Operator integrated with StarlingX 8.0' paragraph
- Later on describe how the platform is more feature rich due to providing support for the aforementioned Intel CPU. For instance: 'By providing support for this hardware platform, StarlingX gains capabilities that makes it a great fit for vRAN deployments. The platform can now support feature X...' or 'By using the Intel vRAN Boost Kubernetes Operator in StarlingX, it enables the platform to ...'

Author: Thank you very much for your review comments and appreciate your thoughts on restructuring the content, however team prefers to keep the original content.

Reviewer: Thank you for getting back to reflect on my comment. Please allow me to provide more context to why I was making the above suggestions.

This a community blog of the StarlingX open source project. The purpose of this blog is to provide content for users and contributors to better understand how the StarlingX project works, how it is or can be used, get the latest news and updates about the project, and learn more about the community. Publishing content that reads as a sales pitch of a commercial product is out of scope for this blog.

In its current form, this blog post has a very strong focus on the capabilities of 4th Intel Xeon Scalable processors, and reads as a sales material for the commercial product. The above suggestions target to update the article, and make it a better fit for the blog.

Does this help clarify the above comment and suggestions?

Author: Thank you for clarification. I am working with internal team for any modification based on your latest suggestions.

Reviewer: Hi @MouliBurla, Are you still planning to upload a new version of this blog post?

### Conclusion

The integration of Intel vRAN Boost Kubernetes Operator in StarlingX provides a ready-made solution for vRAN deployments based on StarlingX Cloud Platform and 4<sup>th</sup> Gen Intel Xeon Scalable processors with Intel vRAN Boost. System administrators can manage the configurability of Intel vRAN Boost accelerator through Kubernetes native CRD APIs and can manage the accelerator’s virtual resource allocation to the vRAN DU application workloads in container platform.

For the complete list of updates and new features in StarlingX 8.0, check out the [release notes](https://docs.starlingx.io/releasenotes/r8-0-release-notes-6a6ef57f4d99.html) and the [project documentation](https://docs.starlingx.io/).

Visit the StarlingX [website](https://www.starlingx.io/) today for further information about the project, check out the [code](https://opendev.org/starlingx), or download the [latest image](https://mirror.starlingx.cengn.ca/mirror/starlingx/release/) to try out the new features.

[1]: For workloads and configurations visit www.Intel.com/PerformanceIndex. Results may vary.

> >Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries.

Author: Was there any specific reason for this suggestion to remove the trademark disclaimer.  Intel legal suggest keeping this line in publications.

> >Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries.

Reviewer: I double checked, it's ok to keep this line here.

I would suggest to use the following formatting:

`**_NOTE:_**` Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries.

> >Intel, the Intel logo, and other Intel marks are trademarks of Intel Corporation or its subsidiaries.

Author: ok, Updated the note as suggested. thanks.
